src/portfft/dispatcher/workitem_dispatcher.hpp

```diff
@@ -140,24 +144,67 @@ PORTFFT_INLINE void workitem_impl(const T* input, T* output, const T* input_imag
   auto loc_load_modifier_view = detail::padded_view(loc_load_modifier, BankLinesPerPad);
   auto loc_store_modifier_view = detail::padded_view(loc_store_modifier, BankLinesPerPad);
 
-  for (IdxGlobal i = global_id; i < round_up_to_multiple(n_transforms, static_cast<IdxGlobal>(SubgroupSize));
-       i += global_size) {
-    bool working = i < n_transforms;
-    Idx n_working = sycl::min(SubgroupSize, static_cast<Idx>(n_transforms - i) + subgroup_local_id);
-
-    IdxGlobal global_offset = static_cast<IdxGlobal>(n_io_reals) * (i - static_cast<IdxGlobal>(subgroup_local_id));
-    if (LayoutIn == detail::layout::PACKED) {
-      if (storage == complex_storage::INTERLEAVED_COMPLEX) {
-        global_data.log_message_global(__func__, "loading non-transposed data from global to local memory");
-        global2local<level::SUBGROUP, SubgroupSize>(global_data, input, loc_view, n_reals * n_working, global_offset,
-                                                    local_offset);
+  const IdxGlobal transform_idx_begin = static_cast<IdxGlobal>(global_data.it.get_global_id(0));
+  const IdxGlobal transform_idx_step = static_cast<IdxGlobal>(global_data.it.get_global_range(0));
+  const IdxGlobal transform_idx_end = round_up_to_multiple(n_transforms, static_cast<IdxGlobal>(SubgroupSize));
+  for (IdxGlobal i = transform_idx_begin; i < transform_idx_end; i += transform_idx_step) {
+    const bool working = i < n_transforms;
+    IdxGlobal leader_i = i - static_cast<IdxGlobal>(subgroup_local_id);
+
+    Idx n_working = sycl::min(SubgroupSize, static_cast<Idx>(n_transforms - leader_i));
+    IdxGlobal global_offset = static_cast<IdxGlobal>(n_io_reals) * leader_i;
+    IdxGlobal global_input_offset = static_cast<IdxGlobal>(input_distance_in_reals) * leader_i;
+    IdxGlobal global_output_offset = static_cast<IdxGlobal>(output_distance_in_reals) * leader_i;
+
+    // This is checking for LayoutIn is PACKED or UNPACKED but we don't actually ever launch kernels with LayoutIn
+    // as UNPACKED.
+    if (LayoutIn == detail::layout::PACKED || LayoutIn == detail::layout::UNPACKED) {
```

Comment:
Our kernels will soon become very hard to follow, with the upcoming subgroup and workgroup strided PRs.
Would it be, or rather would it be a good idea to isolate, batch_interleaved / packed / unpacked logic into different functions, and place them in a different file ?